clang/test/CodeGenSYCL/intel-fpga-local.cpp

```diff
@@ -15,9 +15,15 @@
 //CHECK: [[ANN7:@.str[\.]*[0-9]*]] = {{.*}}{memory:MLAB}
 //CHECK: [[ANN8:@.str[\.]*[0-9]*]] = {{.*}}{memory:DEFAULT}{bankwidth:8}
 //CHECK: [[ANN9:@.str[\.]*[0-9]*]] = {{.*}}{memory:DEFAULT}{max_private_copies:4}
+//CHECK: [[ANN16:@.str.[0-9]*]] = {{.*}}foobar
 
 //CHECK: @llvm.global.annotations
-//CHECK-SAME: a_one{{.*}}[[ANN1]]{{.*}}i32 148
+//CHECK-SAME: @_ZZ3quxiE5a_one
+//CHECK-NOT: to i8*
```

Comment:
It is probably better to check for what you expect to find instead of what you don't expect. There should be only one expected result, right?

Reply:
I am sure that I should have both checks. CHECK-NOT expects that IR will not contain address space cast to i8*, as for globals it will be just a bitcast to i8 addrspace(1)*

Reply:
What I'm trying to say is that if you expect a bitcast to i8 addrspace(1)*, then you should check for a bitcast to i8 addrspace(1)*.

Reply:
I see. It's a bit complicated, maybe I used not the best approach to check and describe this.
valid code:
` i8 addrspace(1)* bitcast (i32 addrspace(1)* @_ZZ3fooiE5a_one to i8 addrspace(1)*)`
invalid code, in turn, contains valid code and cast it to invalid address space:
` i8* addrspacecast (i8 addrspace(1)* bitcast (i32 addrspace(1)* @_ZZ3fooiE5a_one to i8 addrspace(1)*) to i8*)`
So I can't check for a bitcast to i8 addrspace(1) as it is present in addrspace cast too.